spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/cloud_manager_spec.rb

```diff
@@ -0,0 +1,56 @@
+describe ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager do
+  context ".raw_connect" do
+    let(:config) do
+      {
+        :provider               => "IbmCloud",
+        :IbmCloud_project         => "project",
+        :IbmCloud_json_key_string => "encrypted",
+        :app_name               => Vmdb::Appliance.PRODUCT_NAME,
+        :app_version            => Vmdb::Appliance.VERSION,
+      }
+    end
+  end
+  
+  it ".ems_type" do
+    expect(described_class.ems_type).to eq('ibm_cloud_power_virtual_servers')
+  end
+
+  it ".description" do
+    expect(described_class.description).to eq('IBM Power Systems Virtual Servers')
+  end
+
+  it "does not create orphaned network_manager" do
+    ems = FactoryBot.create(:ems_ibm_cloud_power_virtual_servers_cloud)
+    same_ems = ExtManagementSystem.find(ems.id)
+
+    ems.destroy
+    expect(ExtManagementSystem.count).to eq(0)
+
+    same_ems.save!
+    expect(ExtManagementSystem.count).to eq(0)
+  end
+
+  it "moves the network_manager to the same zone as the cloud_manager" do
+    zone1 = FactoryBot.create(:zone)
+    zone2 = FactoryBot.create(:zone)
+
+    ems = FactoryBot.create(:ems_ibm_cloud_power_virtual_servers_cloud, :zone => zone1)
+    expect(ems.network_manager.zone).to eq zone1
+    expect(ems.network_manager.zone_id).to eq zone1.id
+
+    ems.zone = zone2
+    ems.save!
+    ems.reload
+
+    expect(ems.network_manager.zone).to eq zone2
+    expect(ems.network_manager.zone_id).to eq zone2.id
```

Comment:
I'm taking a closer look at this locally - looks like something we might need to fix in the manager classes.

Reply:
https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/12 fixes this